lib/fs/filesystem.go

```diff
@@ -22,23 +22,38 @@ type Filesystem interface {
 	DirNames(name string) ([]string, error)
 	Lstat(name string) (FileInfo, error)
 	Mkdir(name string, perm FileMode) error
+	MkdirAll(name string, perm FileMode) error
```

Comment:
I'm still thinking about that. The real problem, imho, is `OpenFile` which takes a bunch of flags and stuff and is very *file*-specific. That is, does it make sense to be able to send append-only and flags like that? Maybe we can restrict the possibilities